src/components/filter/Filter.tsx

```diff
@@ -29,21 +30,80 @@ type Props = {
 export const Filter = ({
   filterTypes,
   initialFilters,
+  initialStartDate,
   isOverlay = false,
   queryVariables,
   setQueryVariables,
   withSearch = false
 }: Props) => {
-  const [filters, setFilters] = useState<FilterProps>(queryVariables);
+  const [filters, setFilters] = useState<FilterProps>(
+    initialStartDate
+      ? Object.fromEntries(
+          Object.entries(queryVariables).filter(
+            ([key, value]) => !(key === 'start_date' && value === initialStartDate)
+          )
+        )
+      : queryVariables
+  );
   const [isCollapsed, setIsCollapsed] = useState(true);
   const [filterCount, setFilterCount] = useState(0);
 
   useEffect(() => {
     if (!isOverlay) {
-      setQueryVariables((prev) => ({ search: prev.search || '', ...filters }));
+      setQueryVariables((prev) => {
+        const newFilters = { ...filters };
+
+        if (newFilters.start_date === initialStartDate) {
+          delete newFilters.start_date;
+          return {
+            ...prev,
+            search: prev.search || '',
+            start_date: initialStartDate,
+            ...newFilters
+          };
+        }
+
+        if (!newFilters.start_date) {
+          return {
+            ...prev,
+            search: prev.search || '',
+            start_date: initialStartDate,
+            ...newFilters
+          };
+        }
+
+        return { ...prev, search: prev.search || '', ...newFilters };
+      });
     }
   }, [filters]);
 
+  const resetFilters = () => {
+    if (!isOverlay) {
+      setIsCollapsed(!isCollapsed);
+
+      setTimeout(() => {
+        setFilters(
+          initialStartDate
+            ? Object.fromEntries(
+                Object.entries(queryVariables).filter(
+                  ([key, value]) => !(key === 'start_date' && value === initialStartDate)
+                )
+              )
+            : queryVariables
```

Comment:
duplicate content here, as we have this already above.
could this be simplified? why do we need to do object from entries and object entries.
what is the goal here? to filter out a start date if it is the initial one?

Reply:
Yes, that's right, there is a repetition here. I will create a helper function for this.

If the `start_date` value is `initialStartDate`, then remove it from the filter object. This way we prevent the `initialStartDate` date from appearing in the `start_date` input.

Reply:
ok. i understand but this is not what you do. you are not removing something from the filter object. you are iteration multiple times to filter everything else except the initial start date. can you find a way that matches your sentence and translates it into code?

Reply:
still `Object.fromEntries(` and `Object.entries...filer...`
and everytime `queryVariables` changes that line will be run again? you only want it for setting the filters initial state, right?

Reply:
That's right. We just want the filter to be that way to start with. So what do you suggest?

Reply:
I updated the code in this way. Yes `Object...` is still in the code. I couldn't think of any other method.

[171b753](https://github.com/smart-village-solutions/smart-village-app-app/pull/1252/commits/171b75321c0c919670bb73f35fc32a6465171a21)

Reply:
I think it's better now.

[5bebc99](https://github.com/smart-village-solutions/smart-village-app-app/pull/1252/commits/5bebc9965c1c5c3c4ba0f98dc243729ca5439848)